tests/constructor/test_config_structure.py

```diff
@@ -0,0 +1,56 @@
+import sys
+sys.path.append('../../')
+
+import hydra
+from dataclasses import dataclass, field
+from enum import Enum
+from typing import *
+
+from hydra.core.config_store import ConfigStore
+from omegaconf import OmegaConf
+
+from src.constructor.config_structure import ConfigParams, Phase
+
+
+@hydra.main(config_path="configs/", config_name="config.yaml")
+def test_config_load_when_full_config_was_define(config):
+    config_params = ConfigParams(**config)
+    # test if metric phase is enum
+    if isinstance(type(config_params.metrics[0].phases[0]), Phase):
+        print('Failed. Phase is not Enum.')
+    else:
+        print('Phase enum OK.')
+    # test normalize_weights parameter
+    if config_params.losses.normalize_weights != False:
+        print('Failed. normalize_weights != False.')
+    else:
+        print('normalize_weights OK.')
+
+
+@hydra.main(config_path="configs/", config_name="config_without_metrics.yaml")
+def test_config_load_when_metric_is_not_define(config):
```

Comment:
Maybe you should just leave one giant test as `test_config_load_when_full_config_was_defined` or you can split this one by many small tests where every group of parameters is checked independently, so that it will be easier for contributors to find errors